docker/flink/Dockerfile

```diff
@@ -1,3 +1,12 @@
 FROM flink:1.18.1
+
 # ibis-flink requires PyFlink dependency
 RUN wget -nv -P $FLINK_HOME/lib/ https://repo1.maven.org/maven2/org/apache/flink/flink-python/1.18.1/flink-python-1.18.1.jar
+
+# install python3 and pip3
+RUN apt-get update -y && \
+apt-get install -y python3 python3-pip python3-dev && rm -rf /var/lib/apt/lists/*
+RUN ln -s /usr/bin/python3 /usr/bin/python
+
+# install PyFlink
+RUN pip3 install apache-flink==1.18.1
```

Comment:
Actually, let me check if this bit is necessary.

Reply:
Did this end up being necessary?

Reply:
It is.

Reply:
Not blocking the PR, but is this dependency on the server side inevitable for the foreseeable future?

Reply:
I believe so. I think this is similar to how something like Spark also works, where you have to make any Python libraries/files needed by UDFs available to the cluster. Otherwise, you have to submit to the cluster when running, using something like https://nightlies.apache.org/flink/flink-docs-master/docs/dev/python/python_config/#python-options